bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoCrockpotHandler.java

```diff
@@ -0,0 +1,323 @@
+                    mode = "50";
+                    break;
+                case "LOW":
+                    mode = "51";
+                    break;
+                case "HIGH":
+                    mode = "52";
+                    break;
+            }
+            try {
+                String soapHeader = "\"urn:Belkin:service:basicevent:1#SetBinaryState\"";
+                String content = "<?xml version=\"1.0\"?>"
+                        + "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
+                        + "<s:Body>" + "<u:SetCrockpotState xmlns:u=\"urn:Belkin:service:basicevent:1\">" + "<mode>"
+                        + mode + "</mode>" + "<time>" + time + "</time>" + "</u:SetCrockpotState>" + "</s:Body>"
+                        + "</s:Envelope>";
+                String wemoURL = getWemoURL("basicevent");
+
+                if (wemoURL != null) {
+                    wemoHttpCaller.executeCall(wemoURL, soapHeader, content);
+                }
+            } catch (Exception e) {
+                logger.error("Failed to send command '{}' for device '{}': {}", command, getThing().getUID(),
+                        e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+            }
+            updateStatus(ThingStatus.ONLINE);
+        }
+    }
+
+    @Override
+    public void onServiceSubscribed(String service, boolean succeeded) {
+        logger.debug("WeMo {}: Subscription to service {} {}", getUDN(), service, succeeded ? "succeeded" : "failed");
+        subscriptionState.put(service, succeeded);
+    }
+
+    @Override
+    public void onValueReceived(String variable, String value, String service) {
+        logger.debug("Received pair '{}':'{}' (service '{}') for thing '{}'",
+                new Object[] { variable, value, service, this.getThing().getUID() });
+
+        updateStatus(ThingStatus.ONLINE);
+        this.stateMap.put(variable, value);
+    }
+
+    private synchronized void onSubscription() {
+        if (service.isRegistered(this)) {
+            logger.debug("Checking WeMo GENA subscription for '{}'", this);
+
+            String subscription = "basicevent1";
+
+            if ((subscriptionState.get(subscription) == null) || !subscriptionState.get(subscription).booleanValue()) {
+                logger.debug("Setting up GENA subscription {}: Subscribing to service {}...", getUDN(), subscription);
+                service.addSubscription(this, subscription, SUBSCRIPTION_DURATION);
+                subscriptionState.put(subscription, true);
+            }
+
+        } else {
+            logger.debug("Setting up WeMo GENA subscription for '{}' FAILED - service.isRegistered(this) is FALSE",
+                    this);
+        }
+    }
+
+    private synchronized void removeSubscription() {
+        logger.debug("Removing WeMo GENA subscription for '{}'", this);
```

Comment:
It seems that this class does not have toString implementation, however it will be called when logging.

Reply:
There is no toString() needed, this is done within the logger and it is working in all WeMo handlers since initial contribution.

Reply:
okay